conans/client/detect.py

```diff
@@ -54,42 +54,53 @@ def _clang_compiler(output, compiler_exe="clang"):
         return None
 
 
+def _visual_compiler_cygwin(output, version):
+    if os.path.isfile("/proc/registry/HKEY_LOCAL_MACHINE/SOFTWARE/Microsoft/Windows/CurrentVersion/ProgramFilesDir (x86)"):
+        is_64bits = True
+    else:
+        is_64bits = False
+
+    if is_64bits and platform.machine().endswith('64'):
```

Comment:
Some of these lines might improve now, but I was not very sure about the logic when cygwing